docs/DeveloperGuide.md

```diff
@@ -429,6 +426,63 @@ _{more aspects and alternatives to be added}_
 
 _{Explain here how the data archiving feature will be implemented}_
 
+### \[Proposed\] Deadline validation between Tasks and Projects
+
+#### Proposed Implementation
+
+Currently, when a user enters a deadline for a task for a given project, any possible date can be entered including deadlines past the deadline of the project itself.
+
+We recognise that there is always a probability that tasks still can be added beyond a project's deadline, so we will not prevent users from doing so, but this scenario is not common.
+
+Beyond the possibility of users keying in a task deadline that is past the project deadline on purpose, there is also the issue that it may be accidental.
+
+Due to the aforementioned reasons, we believe the project deadlines and task deadlines should be validated.
+
+To be more specific, when tasks that have deadlines past the project deadline are added, there should be a warning indicating that a task with a deadline past the project deadline was added.
+
+Given below is an example usage scenario and how the validation mechanism behaves at each step.
+
+Step 1. The user executes `addT` to add a task to a project with a deadline past the project deadline.
+
+Step 2. The task is added as per normal in the storage, but a warning is also displayed in the `ResultDisplay`
+
```

Comment:
As dicussed, we do not have to go into too much details. Also, these should be at the end of the DG in an appendix titled "Planned Enhancements" as mentioned [here](https://nus-cs2103-ay2324s1.github.io/website/admin/tp-w12.html#2-start-fixing-ped-bugs-before-the-tutorial). Going into too much details introduces risks of other documentation bugs, so it is best to stick to their example format. 

![image](https://github.com/AY2324S1-CS2103T-T08-3/tp/assets/65384645/7bf7560a-2533-408f-8224-196e7fa54790)

Since I am working on this part as well, let me know if you are going to move these to the appendix `Planned Enhancements` . If not, I can approve, merge, and move it myself.